discovery-provider/src/queries/trending.py

```diff
@@ -12,160 +12,34 @@
-                Follow.followee_user_id,
-                func.count(Follow.followee_user_id)
-            )
-            .filter(
-                Follow.is_current == True,
-                Follow.is_delete == False,
-                Follow.followee_user_id.in_(track_owner_list)
-            )
-            .group_by(Follow.followee_user_id)
-            .all()
-        )
-        follower_count_dict = \
-                {user_id: follower_count for (user_id, follower_count) in follower_counts}
-
-        save_counts = get_save_counts(session, False, True, not_deleted_track_ids, None)
-        track_save_counts = {
-            save_item_id: save_count
-            for (save_item_id, save_count, save_type) in save_counts
-            if save_type == SaveType.track
-        }
-
-        trending_tracks = []
-        for track_entry in listen_counts:
-            # Skip over deleted tracks
-            if (track_entry[response_name_constants.track_id] not in not_deleted_track_ids):
-                continue
-
-            # Populate repost counts
-            if track_entry[response_name_constants.track_id] in track_repost_counts:
-                track_entry[response_name_constants.repost_count] = \
-                        track_repost_counts[track_entry[response_name_constants.track_id]]
-            else:
-                track_entry[response_name_constants.repost_count] = 0
-
-            # Populate save counts
-            if track_entry[response_name_constants.track_id] in track_save_counts:
-                track_entry[response_name_constants.save_count] = \
-                        track_save_counts[track_entry[response_name_constants.track_id]]
-            else:
-                track_entry[response_name_constants.save_count] = 0
-
-            # Populate listen counts
-            owner_id = track_owner_dict[track_entry[response_name_constants.track_id]]
-            owner_follow_count = 0
-            if owner_id in follower_count_dict:
-                owner_follow_count = follower_count_dict[owner_id]
-            track_entry[response_name_constants.track_owner_id] = owner_id
-            track_entry[response_name_constants.track_owner_follower_count] = owner_follow_count
-
-            trending_tracks.append(track_entry)
-
-    final_resp = {}
-    final_resp['listen_counts'] = trending_tracks
+        final_resp = generate_trending(get_db(), time, genre, limit, offset)
+        return api_helpers.success_response(final_resp)
+    else:
+        redis_key = f"trending-{time}"
+        redis_cache_value = REDIS.get(redis_key)
+        logger.error(redis_cache_value)
+        if redis_cache_value is not None:
+            json_cache = json.loads(redis_cache_value.decode('utf-8'))
+            if json_cache is not None:
+                num_cached_entries = len(json_cache['listen_counts'])
+                if limit <= num_cached_entries:
+                    json_cache['listen_counts'] = json_cache['listen_counts'][offset:limit]
```

Comment:
@hareeshnagaraj shouldn't this be [offset:offset+limit]?

Reply:
yes so that calculation of limit should also be offset + limit ?

Reply:
yeah cause slicing is [start:end]. if you have an offset of 1000 and a limit of 5, right now this tries to access [1000:5], which won't return anything. it should be [1000:1005]. the second argument is not the number of elements to return, which is really confusing cause